frontend/src/app/features/dashboard/task-info/task-info.component.ts

```diff
@@ -0,0 +1,44 @@
+import * as dashboardActions from '@store/actions/dashboard.actions';
+
+import { Component, OnInit } from '@angular/core';
+
+import { BaseComponent } from '@shared/components/base/base.component';
+import { Store } from '@ngrx/store';
+import { TaskBlock } from '#types/dashboard/timeline/task-block';
+import { selectChosenTaskBlock } from '@store/selectors/dashboard.selectors';
+
+@Component({
+    selector: 'so-task-info',
+    templateUrl: './task-info.component.html',
+    styleUrls: ['./task-info.component.sass']
+})
+export class TaskInfoComponent extends BaseComponent implements OnInit {
+    public taskBlock!: TaskBlock;
+
+    constructor(private store: Store) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.store.select(selectChosenTaskBlock)
```

Comment:
It is ok for me, but here you may easily see an example of what we use in our project.
https://www.npmjs.com/package/ng-let

Instead of subscription and saving result into field, you may have as everywhere just Observable<TaskBlock>. Make only select without pipe and subscription.

In html you only need to write *ngLet="taskBlock$ | async as taskBlock".
For method getBlockStatus from html you will just call {{ getBlockStatus(taskBlock) }}. Also, method will lose dependency from this.taskBlock - it will become parameter.

Reply:
Fixed